commands/app.go

```diff
@@ -153,6 +159,23 @@ To get a linked authority token:
 	return nil
 }
 
+func pathToConfigFile(ctx *cli.Context) (string, error) {
+	if !ctx.IsSet("context") {
+		// no context specified; make sure the user has specified config file
+		if ctx.NArg() == 0 {
+			return "", cli.ShowAppHelp(ctx)
+		}
+
+		return ctx.Args().Get(0), nil
+	}
+
+	if ctx.NArg() > 0 {
+		return "", errors.New("either the context flag or a config file may be specified but not both")
+	}
```

Comment:
I'm not sure about this, although we don't have full support for some scenarios yet, I think it makes sense to allow to specify a ca.json too. 

Some files in the ca.json support relative paths, and they are relative to your current context or STEPPATH if contexts are not being used. If we add support for it for "root", "crt", "key", and I'm perhaps others, it will make total sense to have two different ca.json in the context to try different configurations.

Reply:
I like the idea of being able to select a non-default `ca.json` file for the context. I've had situations where I've wanted to do this.

Reply:
Wouldn't it make it more complex for users, though? Like yet another option to make the configuration more flexible? I create a new CA in a new context if I need to quickly test something, so that I don't have to think about all of the paths involved. I do see some utility in being able to use the same root/intermediate and such, though.

Showing current configuration information at startup time becomes more of a necessity if we were to support this.